stock_release_channel_plan_shipment_lead_time/models/stock_release_channel.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+from odoo import api, fields, models
+
+
+class StockReleaseChannel(models.Model):
+    _inherit = "stock.release.channel"
+
+    delivery_weekday_ids = fields.Many2many(
+        "time.weekday",
+        "release_channel_delivery_weekday_rel",
+        "channel_id",
+        "weekday_id",
+        string="Delivery weekdays",
+    )
+    preparation_weekday_ids = fields.Many2many(
+        "time.weekday",
+        compute="_compute_preparation_weekday_ids",
+        readonly=True,
+        store=True,
+    )
+
+    @api.depends("delivery_weekday_ids", "shipment_lead_time")
+    def _compute_preparation_weekday_ids(self):
+        """Preparation weekdays are delivery weekdays - lead time"""
+        for channel in self:
+            weekday_names = []
+            delivery_weekdays = channel.delivery_weekday_ids
+            for wd in delivery_weekdays:
+                if channel.warehouse_id.calendar_id:
+                    # To consider days off in the calendar, we look for the next date
+                    #  from today on whose weekday matches with current delivery weekday.
+                    # We then deduce the lead time using the calendar to find a weekday
+                    #  that is open.
+                    # It's not perfect, but it's less wrong than only deducing the days
+                    #  without consideration of closed days on weekends
```

Comment:
It looks the best approach. I would have dropped those 2 lines.
Maybe you add this note
```
As this serves for helping in configuring the static preparation plan, we do not consider here the leaves
```